### ISSUE TITLE
Producing Imagenet_classes.txt labels

### ISSUE THREAD

**Issue by EverLookNeverSee** [pytorch/pytorch#46160]
Hi all
I just implemented code below to test resnet101 pre-trained model:

```python
from torchvision import models
from torchvision import transforms
from PIL import Image
import torch

print(dir(models))

alexnet = models.AlexNet()
resnet = models.resnet101(pretrained=True)

print(resnet)

preprocess = transforms.Compose([
    transforms.Resize(256),
    transforms.CenterCrop(224),
    transforms.ToTensor(),
    transforms.Normalize(
        mean=[0.485, 0.456, 0.406],
        std=[0.229, 0.224, 0.225]
    )
])


img = Image.open("sample_dog.jpg")
img_t = preprocess(img)

batch_t = torch.unsqueeze(img_t, 0)

resnet.eval()

out = resnet(batch_t)
print(out)

with open("imagenet_classes.txt") as f:
    labels = [line.strip() for line in f.readline()]

_, index = torch.max(out, 1)


percentage = torch.nn.functional.softmax(out, dim=1)[0] * 100
labels[index[0]], percentage[index[0]].item()
```

but code above does not produce labels.

cc @fmassa @vfdev-5

**Comment by glaringlee** [comment]
@EverLookNeverSee
Did you check 'imagenet_classes.txt' to make sure it has data in it?

**Comment by vfdev-5** [comment]
@EverLookNeverSee take a look here : https://github.com/pytorch/vision/blob/master/examples/python/tensor_transforms.ipynb

```python
import json
from torchvision.datasets.utils import download_url

from torchvision import models
from torchvision import transforms
from PIL import Image
import torch


download_url("https://s3.amazonaws.com/deep-learning-models/image-models/imagenet_class_index.json", ".", "imagenet_class_index.json")


with open("imagenet_class_index.json", "r") as h:
    labels = json.load(h)


alexnet = models.AlexNet()
resnet = models.resnet101(pretrained=True)


preprocess = transforms.Compose([
    transforms.Resize(256),
    transforms.CenterCrop(224),
    transforms.ToTensor(),
    transforms.Normalize(
        mean=[0.485, 0.456, 0.406],
        std=[0.229, 0.224, 0.225]
    )
])


download_url("https://farm1.static.flickr.com/152/434505223_8d1890e1e2.jpg", ".", "sample_dog.jpg")
img = Image.open("sample_dog.jpg")
img_t = preprocess(img)

batch_t = torch.unsqueeze(img_t, 0)

resnet.eval()

with torch.no_grad():
    out = resnet(batch_t)


label_indices = out.argmax(dim=1)
for p in label_indices:
    print(labels[str(p.item())])

> ['n02113023', 'Pembroke']
```

HTH

**Comment by EverLookNeverSee** [comment]
> @EverLookNeverSee take a look here : https://github.com/pytorch/vision/blob/master/examples/python/tensor_transforms.ipynb
> 
> ```python
> import json
> from torchvision.datasets.utils import download_url
> 
> from torchvision import models
> from torchvision import transforms
> from PIL import Image
> import torch
> 
> 
> download_url("https://s3.amazonaws.com/deep-learning-models/image-models/imagenet_class_index.json", ".", "imagenet_class_index.json")
> 
> 
> with open("imagenet_class_index.json", "r") as h:
>     labels = json.load(h)
> 
> 
> alexnet = models.AlexNet()
> resnet = models.resnet101(pretrained=True)
> 
> 
> preprocess = transforms.Compose([
>     transforms.Resize(256),
>     transforms.CenterCrop(224),
>     transforms.ToTensor(),
>     transforms.Normalize(
>         mean=[0.485, 0.456, 0.406],
>         std=[0.229, 0.224, 0.225]
>     )
> ])
> 
> 
> download_url("https://farm1.static.flickr.com/152/434505223_8d1890e1e2.jpg", ".", "sample_dog.jpg")
> img = Image.open("sample_dog.jpg")
> img_t = preprocess(img)
> 
> batch_t = torch.unsqueeze(img_t, 0)
> 
> resnet.eval()
> 
> with torch.no_grad():
>     out = resnet(batch_t)
> 
> 
> label_indices = out.argmax(dim=1)
> for p in label_indices:
>     print(labels[str(p.item())])
> 
> > ['n02113023', 'Pembroke']
> ```
> 
> HTH

Thanks

**Comment by vfdev-5** [comment]
Let me close the issue as answered, feel free to reopen if needed more help on this topic.

**Comment by ylevental** [comment]
> > @EverLookNeverSee take a look here : https://github.com/pytorch/vision/blob/master/examples/python/tensor_transforms.ipynb
> > ```python
> > import json
> > from torchvision.datasets.utils import download_url
> > 
> > from torchvision import models
> > from torchvision import transforms
> > from PIL import Image
> > import torch
> > 
> > 
> > download_url("https://s3.amazonaws.com/deep-learning-models/image-models/imagenet_class_index.json", ".", "imagenet_class_index.json")
> > 
> > 
> > with open("imagenet_class_index.json", "r") as h:
> >     labels = json.load(h)
> > 
> > 
> > alexnet = models.AlexNet()
> > resnet = models.resnet101(pretrained=True)
> > 
> > 
> > preprocess = transforms.Compose([
> >     transforms.Resize(256),
> >     transforms.CenterCrop(224),
> >     transforms.ToTensor(),
> >     transforms.Normalize(
> >         mean=[0.485, 0.456, 0.406],
> >         std=[0.229, 0.224, 0.225]
> >     )
> > ])
> > 
> > 
> > download_url("https://farm1.static.flickr.com/152/434505223_8d1890e1e2.jpg", ".", "sample_dog.jpg")
> > img = Image.open("sample_dog.jpg")
> > img_t = preprocess(img)
> > 
> > batch_t = torch.unsqueeze(img_t, 0)
> > 
> > resnet.eval()
> > 
> > with torch.no_grad():
> >     out = resnet(batch_t)
> > 
> > 
> > label_indices = out.argmax(dim=1)
> > for p in label_indices:
> >     print(labels[str(p.item())])
> > 
> > > ['n02113023', 'Pembroke']
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > HTH
> 
> Thanks

You are the best
